compiler/src/function/main_function.rs

```diff
@@ -61,18 +61,42 @@ impl<'a, F: PrimeField, G: GroupType<F>> ConstrainedProgram<'a, F, G> {
             {
                 let input_variable = input_variable.get().borrow();
                 let name = input_variable.name.name.clone();
-                let input_option = input.get(&name).ok_or_else(|| {
-                    FunctionError::input_not_found(name.clone(), &function.span.clone().unwrap_or_default())
-                })?;
-                let input_value = self.allocate_main_function_input(
-                    cs,
-                    &input_variable.type_.clone(),
-                    &name,
-                    input_option,
-                    &function.span.clone().unwrap_or_default(),
-                )?;
-
-                // Store a new variable for every allocated main function input
+                let is_const = input_variable.const_;
```

Comment:
Do we use is_const anywhere else? If it is not used then we do not need to create this variable.

Reply:
Done.